### PR TITLE
doriswriter support csv

### PR DESCRIPTION
**Pull request by wunan1210**
## Proposed changes

make doriswriter of DataX support format csv.  Format csv is more simple and faster than format json when data is simple(maybe, i think ...).

add property format: csv/json
add property column_separator: effect when format is csv, for example "\x01" , "^", etc...

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Code refactor (Modify the code structure, format the code, etc...)
- [ ] Optimization. Including functional usability improvements and performance improvements.
- [ ] Dependency. Such as changes related to third-party components.
- [ ] Other.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have created an issue on (Fix #ISSUE) and described the bug/feature there in detail
- [x] Compiling and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If these changes need document changes, I have updated the document
- [ ] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hf200012** [comment]
I don't think there's any need to do a conversion here, and the csv format can cause problems, such as empty strings, with the separator you specify in the data，The json format is the least error-prone

File: extension/DataX/doriswriter/src/main/java/com/alibaba/datax/plugin/writer/doriswriter/DorisCsvCodec.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    @Override
+    public String serialize(final Record row) {
+        if (null == this.fieldNames) {
+            return "";
+        }
+        List<String> list = new ArrayList<>();
+
+        for (int i = 0; i < this.fieldNames.size(); i++) {
+            Object value = this.convertColumn(row.getColumn(i));
+            list.add(value != null ? value.toString() : "");
```
Comment (morningman) [inline extension/DataX/doriswriter/src/main/java/com/alibaba/datax/plugin/writer/doriswriter/DorisCsvCodec.java:47]: for null value, it should be `"\N"`

File: extension/DataX/doriswriter/src/main/java/com/alibaba/datax/plugin/writer/doriswriter/Key.java [hunk 2]
```diff
@@ -22,6 +22,8 @@ Licensed to the Apache Software Foundation (ASF) under one
 import com.alibaba.datax.common.exception.DataXException;
 import com.alibaba.datax.common.util.Configuration;
 import com.alibaba.datax.plugin.rdbms.util.DBUtilErrorCode;
+import com.alibaba.fastjson.JSON;
```
Comment (morningman) [inline extension/DataX/doriswriter/src/main/java/com/alibaba/datax/plugin/writer/doriswriter/Key.java:25]: unused imports?

File: extension/DataX/doriswriter/doc/doriswriter.md [hunk 3]
```diff
@@ -163,6 +163,18 @@ DorisWriter 通过Doris原生支持Stream load方式导入数据， DorisWriter
...
+
+  - 描述：导入数据的格式, 可以使是json或者csv。
+  - 必选：否
+  - 默认值：`json`
+  
+* **columnSeparator**
+
+  - 描述：当导入的格式是csv时, 字段之间的分隔符。
+  - 必选：否
+  - 默认值：`\t`
```
Comment (morningman) [inline extension/DataX/doriswriter/doc/doriswriter.md:177]: There may be any character exist in user data, such as `\t` or `,`.
So it is better to use a invisible multi-bytes separator as default column separator here, I recommend to use `\x0707`

**Comment by morningman** [comment]
> I don't think there's any need to do a conversion here, and the csv format can cause problems, such as empty strings, with the separator you specify in the data，The json format is the least error-prone

I think it ok to offer a option to user. And the default column separator can be multi-bytes separator to avoid some problems

File: extension/DataX/doriswriter/doc/doriswriter.md (hunk 3 above)
Comment (wunan1210) [reply extension/DataX/doriswriter/doc/doriswriter.md:177]: already support invisible multi-bytes, like '\x01\x02'. but in version 0.14,  doris seem to only accept one byte. Branch master looks to handle multi-bytes .  ^_^

File: extension/DataX/doriswriter/src/main/java/com/alibaba/datax/plugin/writer/doriswriter/Key.java (hunk 2 above)
Comment (wunan1210) [reply extension/DataX/doriswriter/src/main/java/com/alibaba/datax/plugin/writer/doriswriter/Key.java:25]: my carelessness ...

File: extension/DataX/doriswriter/doc/doriswriter.md (hunk 3 above)
Comment (wunan1210) [reply extension/DataX/doriswriter/doc/doriswriter.md:177]: You mean like this ?
```
list.add((value == null || value.equals("\\N")) ? "" : value.toString());
```

File: extension/DataX/doriswriter/src/main/java/com/alibaba/datax/plugin/writer/doriswriter/DorisCsvCodec.java (hunk 1 above)
Comment (wunan1210) [reply extension/DataX/doriswriter/src/main/java/com/alibaba/datax/plugin/writer/doriswriter/DorisCsvCodec.java:46]: You mean like this ?
```
list.add((value == null || value.equals("\\N")) ? "" : value.toString());
```

**Comment by wunan1210** [comment]
When in stream load using format csv, does Doris recognize string '' ornull ?
My tests show only insert '' .
Can Doris insert null in this way ? Maybe a limitation when using format csv ?

File: extension/DataX/doriswriter/src/main/java/com/alibaba/datax/plugin/writer/doriswriter/DorisCsvCodec.java (hunk 1 above)
Comment (morningman) [reply extension/DataX/doriswriter/src/main/java/com/alibaba/datax/plugin/writer/doriswriter/DorisCsvCodec.java:46]: Like this: `list.add(value != null ? value.toString() : "\\N");`

File: extension/DataX/doriswriter/doc/doriswriter.md (hunk 3 above)
Comment (morningman) [reply extension/DataX/doriswriter/doc/doriswriter.md:177]: OK I see. So please modify the description to remind this risk.

**Comment by morningman** [comment]
> When in stream load using format csv, does Doris recognize string '' or null ?
> My tests show only insert '' .
> Can Doris insert null in this way ? Maybe a limitation when using format csv ?

For null value, it should be "\N"

**Comment by wunan1210** [comment]
> > When in stream load using format csv, does Doris recognize string '' or null ?
> > My tests show only insert '' .
> > Can Doris insert null in this way ? Maybe a limitation when using format csv ?
> 
> For null value, it should be "\N"

thanks for your comment, it confused me a lot ╰(￣▽￣)╮

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**